src/controllers/user-order.controller.ts

```diff
@@ -0,0 +1,107 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/example-shopping
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {repository, Filter, Where} from '@loopback/repository';
+import {UserRepository} from '../repositories';
+import {
+  post,
+  get,
+  patch,
+  del,
+  param,
+  requestBody,
+  HttpErrors,
+} from '@loopback/rest';
+import {Order} from '../models';
+
+/**
+ * Controller for User's Orders
+ */
+export class UserOrderController {
+  constructor(@repository(UserRepository) protected userRepo: UserRepository) {}
+
+  /**
+   * Create or update the orders for a given user
+   * @param userId User id
+   * @param cart Shopping cart
+   */
+  @post('/users/{userId}/orders', {
+    responses: {
+      '200': {
+        description: 'User.Order model instance',
+        content: {'application/json': {'x-ts-type': Order}},
+      },
+    },
+  })
+  async create(
+    @param.path.string('userId') userId: string,
+    @requestBody() order: Order,
+  ): Promise<Order> {
+    if (userId !== order.userId) {
+      throw new HttpErrors.BadRequest(
+        `User id does not match: ${userId} !== ${order.userId}`,
+      );
+    }
+    delete order.userId;
```

Comment:
I find it suspicious that we have to delete `order.userId`. I would expect the hasMany relation repository to take care of that. What am I missing?

Reply:
Related to https://github.com/strongloop/loopback-next/issues/1719. This is a shortcoming of the relation repository where it actually throws an error if the property is present. If I don't delete the `userId` myself, I get the following error: 

```
Unhandled error in POST /users/5ba12137d078ee5357c02b00/orders: 500 Error: Property "userId" cannot be changed!
    at Object.constrainDataObject (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/repository/src/repositories/constraint-utils.ts:59:13)
    at DefaultHasManyEntityCrudRepository.create (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/repository/src/repositories/relation.repository.ts:79:7)
    at UserOrderController.createOrders (/Users/taranveer/LoopBack/loopback4-example-shopping/src/controllers/user-order.controller.ts:48:47)
    at value_promise_1.transformValueOrPromise.args (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/context/src/resolver.ts:240:37)
    at Object.transformValueOrPromise (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/context/src/value-promise.ts:269:12)
    at Object.invokeMethod (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/context/src/resolver.ts:235:10)
    at ControllerRoute.invokeHandler (/Users/taranveer/LoopBack/loopback4-example-shopping/node_modules/@loopback/rest/src/router/routing-table.ts:418:18)
    at process._tickCallback (internal/process/next_tick.js:68:7)
```